### PR TITLE
Refresh requirements

### PR DESCRIPTION
**Pull request by jakirkham**
* Require Numba 0.57+ consistently
* Require NumPy 1.21+ (to align with [Numba 0.57.0's NumPy minimum]( https://github.com/numba/numba/blob/0.57.0/setup.py#L25 ))
* Allow NumPy 1.24 (soften upper bounds)
* Replace deprecated `np.int` with `int`

### CODE REVIEW

**Review by ajschmidt8 — APPROVED** [review]

File: dependencies.yaml
Comment (bdice) [inline dependencies.yaml:100]: I think we're pinning `numpy >=1.21` in most other RAPIDS packages.

Comment (jakirkham) [reply dependencies.yaml:100]: Since this was already setting a lower bound of NumPy 1.19, went ahead and left it. No strong feelings though. Will defer to maintainers here

Comment (bdice) [reply dependencies.yaml:100]: I'm happy to leave it as-is, too -- just wanted to give some context for why that version is relevant.

Comment (jakirkham) [reply dependencies.yaml:100]: Yeah Numba 0.57.0 is needed for CUDA 12 support

Ah ok. That's a good point. Let's bump NumPy to 1.21 then

File: conda/environments/all_cuda-118_arch-x86_64.yaml
Comment (jakirkham) [inline conda/environments/all_cuda-118_arch-x86_64.yaml:15]:
```suggestion
- numpy >=1.21
```

File: conda/recipes/cusignal/meta.yaml
Comment (jakirkham) [inline conda/recipes/cusignal/meta.yaml:32]:
```suggestion
    - numpy >=1.21
```

File: dependencies.yaml
Comment (jakirkham) [reply dependencies.yaml:100]:
```suggestion
          - numpy >=1.21
```

**Review by jakirkham — COMMENTED** [review]
Bumping NumPy to 1.21 to align with Numba 0.57.0's [NumPy requirement]( https://github.com/numba/numba/blob/0.57.0/setup.py#L25 )

**Review by bdice — APPROVED** [review]

**Review by raydouglass — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks all! 🙏

**Pull request merged**
